### PR TITLE
Add linter and fix lint issues

### PR DESCRIPTION
**Pull request by rahulPalliyalil**
### What

- This adds the golangci linter to the dp-search-reindex-api repo. This PR enables the following linters via .golangci.yml
`
    - deadcode
    - depguard
    - dogsled
    - errcheck
    - gochecknoinits
    - goconst
    - gocritic
    - gocyclo
    - gofmt
    - goimports
    - revive
    - gosec
    - gosimple
    - govet
    - ineffassign
    - nakedret
    - staticcheck
    - structcheck
    - stylecheck
    - typecheck
    - unconvert
    - unused
    - varcheck
    - whitespace
    - gocognit
    - prealloc`

This PR also attempts to fix the existing lint issues with the repo. **Please Note** in .golangci.yaml linter configuration file there is an parameter under issues attribute reffered to as `new` I have set it to false (default), but when you set it to true then it will ignore all the existing lint issues and will flag only the new lint issues. 

### How to review

Please have a look into the .golangcilint.yml file and see if the linters listed here make sense , also if you would like to add any additional linters we can always do that . For the first pass I think we have enough linters enabled (mentioned above).

### CODE REVIEW

**Review by bpathak-ons — COMMENTED** [review]
Looks good to me with few minor comments

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# This file was inspired by the golangci-lint one:
```
Comment (bpathak-ons) [inline .golangci.yml:1]: just wondering if lint is in Makefile and get the output in a file.
currently lint action is doing nothing

File: features/steps/component.go [hunk 2]
```diff
@@ -19,24 +21,28 @@ import (
 const outputFilePath = "/tmp/dpSearchDataExtractor.txt"
```
Comment (bpathak-ons) [inline features/steps/component.go:21]: I think this can be deleted as it was initial implementation

File: features/publish_data.feature [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Feature: Data extractor should listen to the relevant topic and publish extracted data
```
Comment (bpathak-ons) [inline features/publish_data.feature:1]: I think, component tests changes should be in a separate PR

Comment (rahulPalliyalil) [reply features/publish_data.feature:1]: The component test changes is already merged. I think there is an issue with the rebase . Will fix it

File: .golangci.yml (hunk 1 above)
Comment (rahulPalliyalil) [reply .golangci.yml:1]: This is the golangci lint file , that is the configuration file for the linter. This is being taken into consideration when the golangcilint command is run
